vmm/src/memory_manager.rs

```diff
@@ -450,10 +499,32 @@ impl MemoryManager {
         Ok(slot)
     }
 
+    pub fn virtiomem_resize(&mut self, size: u64) -> Result<(), Error> {
+        let region = self.virtiomem_region.take();
+        if let Some(region) = region {
+            self.add_region(region)?;
+        }
+
+        if let Some(resize) = &self.virtiomem_resize {
+            resize.work(size).map_err(Error::VirtioMemResizeFail)?;
+        } else {
+            panic!();
```

Comment:
I use panic in both places because if executed in these two places it means something is wrong with the code.  For example:
https://github.com/cloud-hypervisor/cloud-hypervisor/blob/d8d790bb7bf5a724de383011015b0a543d2ba875/vhost_user_fs/src/descriptor_utils.rs#L594

What about I change this part to:
panic!("should not fail here")

Reply:
I agree if this fails, then something is wrong with the code, but I still think returning an error is better as this means it can be properly handled and propagated all the way.

Reply:
I think If the developer does something wrong.  I think let the application stop at once is better.
Because error code can break the user's system.  It may also destroy important date or introduce a system vulnerability.

Reply:
Ok so in this case, because the panic is going to be called from from the main thread, I'm okay if you want to keep it.
But in the `vm-virtio/src/mem.rs` you have to be careful when using `panic()`. If the code runs in a different thread, the thread is going to be killed, but all the other thread will continue. This is a really bad behavior as the VMM might still be running while the virtio-mem thread won't be there anymore.

Reply:
panic! in mem.rs is only called by MemoryManager.resize->MemoryManager.virtiomem_resize.
Do you think I can keep it?